opm/grid/utility/cartesianToCompressed.cpp

```diff
@@ -23,31 +24,30 @@
 namespace Opm
 {
 
-    // Construct explicit mapping from local cartesian to active/compressed
-    // indices, either based on global_cell or as { 0, 1, 2, ....} if null.
-    // \param[in] num_cartesian_cells    The number of cartesian cells.
-    // \param[in] global_cell  Either null, or an array of size num_cells.
-    // \return                 A unordered_map containing the mapping from local cartesian
-    //                         to active/compressed,
-    //                         or the map { {0, 0}, {1, 1}, ... , {num_cells - 1, num_cells - 1} }
-    //                         if global_cell was null.
-    std::unordered_map<int, int> cartesianToCompressed(const int num_cells,
-                                                       const int* global_cell)
-    {
-        std::unordered_map<int, int> retval;
-        retval.max_load_factor(0.9);
-        retval.reserve(num_cells);
-        if (global_cell) {
-            for (int i = 0; i < num_cells; ++i) {
-                retval.insert({global_cell[i], i});
-            }
-        } else {
-            for (int i = 0; i < num_cells; ++i) {
-                retval.insert({i, i});
-            }
+// Construct explicit mapping from local cartesian to active/compressed
+// indices, either based on global_cell or as { 0, 1, 2, ....} if null.
+// \param[in] num_cartesian_cells    The number of cartesian cells.
+// \param[in] global_cell  Either null, or an array of size num_cells.
+// \return                 A unordered_map containing the mapping from local cartesian
+//                         to active/compressed (Attention! When LGRs, one key takes multiple values),
+//                         or the map { {0, 0}, {1, 1}, ... , {num_cells - 1, num_cells - 1} }
+//                         if global_cell was null.
+std::unordered_multimap<int, int> cartesianToCompressed(const int num_cells,
+                                                        const int* global_cell)
+{
+    std::unordered_multimap<int, int> retval;
+    retval.reserve(num_cells);
+    if (global_cell) {
+        for (int i = 0; i < num_cells; ++i) {
+            retval.insert(std::pair<int,int>(global_cell[i], i));
+        }
+    } else {
+        for (int i = 0; i < num_cells; ++i) {
+            retval.insert(std::pair<int,int>(i, i));
         }
-        return retval;
     }
+    return retval;
```

Comment:
If you're changing this anyway, now would be a good time to switch to
```C++
retval.emplace(x, y)
```
instead of
```C++
retval.insert(std::pair<int,int>(x, y))
```
They do the same thing, effectively, but the former (`.emplace()`) is (slightly) faster since it constructs the value in-place.

Reply:
Thanks for your feedback! Indeed, unordered_map<int, std::vector<int>> is an alternative too. I'll check it out.